packages/hledger/build.sh

```diff
@@ -0,0 +1,29 @@
+TERMUX_PKG_HOMEPAGE=https://hledger.org/
+TERMUX_PKG_DESCRIPTION="Robust, friendly, fast plain text accounting software"
+TERMUX_PKG_LICENSE="GPL-3.0"
+TERMUX_PKG_MAINTAINER="@erplsf"
+TERMUX_PKG_VERSION=1.43.2
+TERMUX_PKG_SRCURL=https://github.com/simonmichael/hledger/archive/refs/tags/${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=60b74c70ddfc6b84ca87debd2ac302aac754da3c0d9089821182e56796cb841e
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_BUILD_DEPENDS="aosp-libs, libffi, libiconv, libgmp, zlib, ncurses"
+
+termux_step_pre_configure() {
+	termux_setup_ghc
+	termux_setup_cabal
+}
+
+termux_step_post_configure() {
+	if [[ "$TERMUX_ON_DEVICE_BUILD" == false ]]; then # We do not need iserv for on device builds.
+		termux_setup_ghc_iserv
+		cat <<-EOF >>cabal.project.local
+			package *
+			    ghc-options: -fexternal-interpreter -pgmi=$(command -v termux-ghc-iserv)
+
+			source-repository-package
+			    type: git
+			    location: https://github.com/erplsf/splitmix
+			    tag: aa5bb94be0d6a3e94584ac0714b2f50237d558db
```

Comment:
That tag value definitely shouldn't be hardcoded.
And that is a repository controlled by you instead of the dependencies upstream repo.

I'm not accusing you of anything nefarious, but the way we usually handle making changes to dependencies is via patches in the package directory that get applied at build time.
https://github.com/termux/termux-packages/blob/b4c1a6d37a36ef18f78a5cf1a1224255826f149a/packages/openethereum/build.sh#L68-L94
This is a Rust example, but hopefully gets the point across.

Reply:
I'm not sure if `cabal` has a concept of "vendoring" a dependency (e.g. using a local copy).

Reply:
Okay, I've tried https://github.com/termux/termux-packages/tree/1367b68bd55484d3366b8a41c73936853456c700/./ `splitmix` and patching it myself following [this](https://github.com/termux/termux-packages/blob/1367b68bd55484d3366b8a41c73936853456c700/packages/pandoc/build.sh#L22-L36) example, but I'm out of luck. I guess we need to wait for @MrAdityaAlok.